pkg/arvo/sys/hoon.hoon

```diff
@@ -262,8 +262,8 @@
 ++  tail  |*(^ ,:+<+)                                   ::  get tail
 ++  test  |=(^ =(+<- +<+))                              ::  equality
 ::
-++  lead  |*(h=* |*(* [h +<]))                          ::  put head
-++  late  |*(t=* |*(* [+< t]))                          ::  put tail
+++  lead  |*(h=* |*(* [+>+< +<]))                       ::  put head
+++  late  |*(t=* |*(* [+< +>+<]))                       ::  put tail
```

Comment:
IMO, you should also remove the `h`/`t` faces from the sample of the outer gate. And it might be clearer to use `+<.^$` over `+>+<`. (Not sure about that last part ...)

Reply:
Good call on removing the faces. To my mild surprise, your stylistic suggestion isn't purely stylistic though:

```hoon
> ? ((lead-lark 2) 3)
  {@ud @ud}
[2 3]
> ? ((lead-ktbc 2) 3)
  {* @ud}
[2 3]
```

Reply:
Wow, that's pretty bad.

```
> `@`(|*(* +<) 1)
1
> `@`(|*(* +<.$) 1)
nest-fail
```

Reply:
Even worse:

```
> `@`(|*(@ +<.$) 1)
1
> `@`(|*(@ +<.$) [1 1])
<<unlinks-dojo>>
```